File: src/test/java/org/cloudfoundry/credhub/util/DatabaseProfileResolver.java
package org.cloudfoundry.credhub.util;

import org.springframework.test.context.support.DefaultActiveProfilesResolver;

import org.apache.commons.lang3.ArrayUtils;

public class DatabaseProfileResolver extends DefaultActiveProfilesResolver {

  @Override
  public String[] resolve(Class<?> testClass) {
    return (String[]) ArrayUtils.addAll(new String[]{System.getProperty(SpringUtilities.activeProfilesString)},
      super.resolve(testClass));
  }
}

File: components/test-support/src/test/kotlin/org/cloudfoundry/credhub/utils/DatabaseProfileResolver.kt
package org.cloudfoundry.credhub.utils

import org.springframework.test.context.support.DefaultActiveProfilesResolver

class DatabaseProfileResolver : DefaultActiveProfilesResolver() {
    override fun resolve(testClass: Class<*>): Array<String> =
        arrayOf(System.getProperty(SpringUtilities.ACTIVE_PROFILE_STRING)).plus(super.resolve(testClass))
}
